### PR TITLE
Chore/cursors updates

### PR DESCRIPTION
**Pull request by ItsWendell**

### CODE REVIEW

**Review by Floris497 — APPROVED** [review]

**Review by Floris497 — APPROVED** [review]

File: srcs/vm/cursors.c [hunk 1]
```diff
@@ -38,35 +38,59 @@ t_cursor    *cursor_clone(t_cursor *origin)
...
 	{
-		prev = del->prev;
-		free(del);
-		current->prev = prev;
+		*head = temp->next;
+		free(temp);
 	}
-	return (current);
+
+    while (temp != NULL && temp->id != id)
```
Comment (Floris497) [inline srcs/vm/cursors.c:83]: Use after free.. please fix

File: srcs/vm/cursors.c [hunk 2]
```diff
@@ -38,35 +38,59 @@ t_cursor    *cursor_clone(t_cursor *origin)
...
+
+    while (temp != NULL && temp->id != id) 
+    {
+        prev = temp;
+        temp = temp->next;
+    }
+
+    if (temp == NULL) return;
+
+    prev->next = temp->next;
```
Comment (Floris497) [inline srcs/vm/cursors.c:91]: prep might not be initialized?

**Review by Floris497 — COMMENTED** [review]
Please fix :)

File: srcs/vm/cursors.c (hunk 1 above)
Comment (ItsWendell) [reply srcs/vm/cursors.c:83]: what?

Comment (Floris497) [reply srcs/vm/cursors.c:83]: You free temp 3 lines up, then do not set it to follow another pointer, so you are using freed memory there..

Comment (ItsWendell) [reply srcs/vm/cursors.c:83]: You see that return statement there? I free temp ONLY if head of the list already is the item id we want to remove. so then we set ->next as the new head and we free temp. saying this does make me realize I made a mistake there, id != id should have been id == id

**Pull request merged**